validation/src/block_production.rs

```diff
@@ -180,12 +168,9 @@ impl<Client, TxPool, Backend> consensus::Proposer<Block> for Proposer<Client, Tx
 			Duration::from_millis(self.slot_duration / SLOT_DURATION_DENOMINATOR),
 		);
 
-		let parent_id = self.parent_id.clone();
-		let client = self.client.clone();
-		let transaction_pool = self.transaction_pool.clone();
 		let table = self.tracker.table().clone();
-		let backend = self.backend.clone();
 		let metrics = self.metrics.clone();
+		let mut proposer = self.proposer.take().expect("Only one block proposed.");
```

Comment:
I have no idea if this assumption is reasonable, if not then probably deferring creation of proposer is necessary. If yes, then it might be a good idea to change the trait to reflect that (`self` vs `&mut self`)